xl: minor fix parsing disk description

Do not set *p = '\0' as it will break detecting if there is r/w parameter.
Although setting 'w' for cdrom is a rare case, still this is a bug.

Signed-off-by: Marek Marczykowski <marmarek@invisiblethingslab.com>
Index: xen-hvm/tools/libxl/xl_cmdimpl.c
===================================================================
--- xen-hvm.orig/tools/libxl/xl_cmdimpl.c
+++ xen-hvm/tools/libxl/xl_cmdimpl.c
@@ -581,8 +581,7 @@ static int parse_disk_config(libxl_devic
             break;
         case DSTATE_VIRTTYPE:
             if ( *p == ',' || *p == '\0' ) {
-                *p = '\0';
-                if ( !strcmp(tok, "cdrom") ) {
+                if ( !strncmp(tok, "cdrom", 5) ) {
                     disk->is_cdrom = 1;
                     disk->unpluggable = 1;
                 }else{
